src/Sylius/Bundle/CoreBundle/Resources/translations/validators.en.yml

```
@@ -44,6 +44,8 @@ sylius:
         enabled:
             cannot_disable_base: The base locale cannot be disabled.
     product:
+        channel:
+            not_available: 'There is no product %productName% in this shop.'
```

I would be for adding this message to the `validators.en.yaml` in `ApiBundle` as the validator is there
```suggestion
            not_available: 'The product %productName% does not exist.'
```